syft/pkg/license.go

```diff
@@ -122,13 +129,35 @@ func NewLicenseFromURLs(value string, urls ...string) License {
 	return l
 }
 
+func stripUnwantedCharacters(rawURL string) (string, error) {
+	// Step 1: Remove newline and tab characters
+	cleanedURL := strings.ReplaceAll(rawURL, "\n", "")
+	cleanedURL = strings.ReplaceAll(cleanedURL, "\t", "")
+
+	// Step 2: Trim leading/trailing spaces
+	cleanedURL = strings.TrimSpace(cleanedURL)
+
+	// Step 3: Validate the cleaned URL
+	_, err := url.ParseRequestURI(cleanedURL)
```

Comment:
should we leave these in the core model but then filter them when encoding sbom specs that have specific requirements? That is, should we move this to the cyclonedx encoder? We could check for any similar requirement for SPDX and do the same.

Reply:
I think it's best to clean it as early as possible in the process and get our own model's URL inline with the RFC 3987 IRI-reference